src/pages/Faucet/components/FaucetModal/FaucetModal.tsx

```diff
@@ -0,0 +1,71 @@
+import { useRef, useState } from 'react';
+
+import { Loader } from 'components';
+import { getEgldLabel } from 'lib';
+import { DataTestIdsEnum } from 'localConstants';
+
+import {
+  useGetFaucetSettingsQuery,
+  useRequestFundsMutation
+} from 'redux/endpoints';
+import { FaucetError } from '../FaucetError';
+import { FaucetScreen } from '../FaucetScreen';
+import { FaucetSuccess } from '../FaucetSuccess';
+
+export const FaucetModal = () => {
+  const ref = useRef(null);
+  const [getFunds, { isSuccess }] = useRequestFundsMutation();
+  const [fundsReceived, setFundsReceived] = useState(false);
+  const [requestFailed, setRequestFailed] = useState(false);
+
+  const { data: settings, error: settingsError } = useGetFaucetSettingsQuery();
+  const egldLabel = getEgldLabel();
+
+  const onRequestClick = async (captcha: string) => {
```

Comment:
Let's rename these `handleRequestClick`. We convened with @arhtudormorar that the `on` prefix is used for callbacks passed as properties of the component.